Program.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+
+namespace Singleton
+{
+    public class Storage
+    {
+        private static int count;
+        private static Storage Instance;
+
+        private Storage()
+        {
+            count++;
+        }
+
+        public void Print()
+        {
+            Console.WriteLine("Storage instance count: " + count);
+        }
+        
+        public static Storage getInstance()
+        {
+            if (Instance != null)
+                return Instance;
+            
+            lock (Instance)
```

Comment:
here you put the instance as the `lock()` argument but instead you should define another object then put it as the `lock()` argument like this
```c#
private static readonly object _lock = new object();
```
then  `lock(_lock)`